php/libraries/SubprojectController.class.inc

```diff
@@ -0,0 +1,254 @@
+<?php
+/**
+ * This file contains the SubprojectController class.
+ *
+ * PHP Version 7
+ *
+ * @category Controller
+ * @package  Subprojects
+ * @author   Rida Abou-Haidar <rida.abou-haidar@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS;
+
+/**
+ * Base class that contains method to access the subproject of projects
+ *
+ * @category Controller
+ * @package  Subprojects
+ * @author   Rida Abou-Haidar <rida.abou-haidar@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class SubprojectController
+{
+    /**
+     * The database connection to use
+     *
+     * @var \Database
+     */
+    protected $database;
+
+    /**
+     * The constructor
+     *
+     * @param \Database $database The database connection to use
+     */
+    public function __construct(\Database $database)
+    {
+        if (!$database->isConnected()) {
```

Comment:
do you need to do that? It's unlikely that tis is the first place an exception will rise if the database is not connected.

Reply:
good point, not sure where I got that from, it might be a leftover from melanie's original code